src/index/functions.rs

```diff
@@ -84,3 +88,56 @@ impl Drop for Index {
         }
     }
 }
+
+#[pgrx::pg_extern(sql = "")]
+fn _vchordrq_logged_queries(
+    indexrelid: Oid,
+) -> pgrx::iter::SetOfIterator<'static, pgrx::composite_type!('static, LOGGED_QUERY_COMPOSITE_TYPE)>
+{
+    use pgrx::heap_tuple::PgHeapTuple;
+
+    let index_oid = if indexrelid == pgrx::pg_sys::InvalidOid {
+        None
+    } else {
+        Some(indexrelid.to_u32())
+    };
+    let queries = unsafe { QueryRecorderMaster::load_all(index_oid) };
+
+    let mut table_cache: HashMap<u32, (String, String)> = HashMap::new();
+    let mut index_cache: HashMap<u32, String> = HashMap::new();
+    let mut dumps = Vec::with_capacity(queries.len());
+    for query in queries {
+        let (namespace, table_name) = match table_cache.get(&query.table_oid).cloned() {
+            Some((namespace, table_name)) => (namespace, table_name),
+            None => match query.search_table() {
+                Some((namespace, table_name)) => {
+                    table_cache.insert(query.table_oid, (namespace.clone(), table_name.clone()));
+                    (namespace, table_name)
+                }
+                None => continue,
+            },
+        };
+        let column_name = match index_cache.get(&query.index_oid).cloned() {
+            Some(column) => column,
+            None => match query.search_column() {
+                Some(column) => {
+                    index_cache.insert(query.index_oid, column.clone());
+                    column
+                }
+                None => continue,
+            },
+        };
+        dumps.push(query.dump(namespace, table_name, column_name));
+    }
+    pgrx::iter::SetOfIterator::new(dumps.into_iter().map(|dump| {
+        let mut view = PgHeapTuple::new_composite_type(LOGGED_QUERY_COMPOSITE_TYPE).unwrap();
```

Comment:
This will search for the type in the current `search_path`, so it is not quite correct.